File: iosApp/iosApp/Preview Content/DesignSystem/TrenifyComponentPreviews.swift
#if DEBUG
import SwiftUI

/// Development-only fixtures. No providers, navigation, KMP state or production gallery entry.
struct TrenifyComponentExamples: View {
    @State private var selection = "journey"

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: TrenifySpacing.xl) {
                TrenifySectionHeader(title: "Component family", subtitle: "Deterministic railway display fixtures")
                TrenifySegmentedControl(options: [TrenifySegment(id: "journey", title: "Journey"), TrenifySegment(id: "train", title: "Train")],
                                        selection: $selection, accessibilityLabel: "Search mode")
                TrenifyStationField(roleLabel: "From", stationName: "San Benedetto del Tronto Porto d’Ascoli",
                                    placeholder: "Choose station", selected: true, onClear: {}) {}
                TrenifyStationField(roleLabel: "To", stationName: nil, placeholder: "Choose station", selected: false, focused: true) {}
                TrenifyStationField(roleLabel: "Station", stationName: "Roma Termini", placeholder: "Choose station", selected: true, enabled: false) {}
                TrenifyPrimaryAction(title: "Search trains") {}
                TrenifyPrimaryAction(title: "Search trains", loading: true) {}
                TrenifySecondaryAction(title: "Change selection", enabled: false) {}
                TrenifyJourneyCard(trainIdentity: "Trenitalia · Frecciarossa 9516", departureTime: "14:10", arrivalTime: "17:19",
                                   origin: "Milano Centrale", destination: "Roma Termini", durationLabel: "3 h 09 min", changesLabel: "Direct",
                                   statusTone: .onTime, statusLabel: "On time", platformLabel: "Scheduled platform 8") {}
                TrenifyStatusPill(tone: .delayed, label: "Delayed +12 min")
                TrenifyStatusPill(tone: .cancelled, label: "Cancelled")
                TrenifyStatusPill(tone: .arrived, label: "Arrived")
                TrenifyStatusPill(tone: .warning, label: "Service warning")
                TrenifyStatusPill(tone: .unknown, label: "Status unavailable")
            }
            .padding(TrenifySpacing.screenHorizontal)
        }
        .background(TrenifyColors.background)
    }
}

struct TrenifyComponentPreviews: PreviewProvider {
    static var previews: some View {
        Group {
            TrenifyComponentExamples().environment(\.colorScheme, .light).previewDisplayName("Light / component family")
            TrenifyComponentExamples().environment(\.colorScheme, .dark).previewDisplayName("Dark / component family")
            TrenifyComponentExamples().environment(\.dynamicTypeSize, .accessibility3)
                .previewLayout(.fixed(width: 320, height: 1800)).previewDisplayName("Large text / narrow")
            TrenifyComponentExamples().environment(\.colorScheme, .dark)
                .environment(\.trenifyAccessibility, TrenifyAccessibility(reduceMotion: true, reduceTransparency: true))
                .previewDisplayName("Dark / opaque actions / reduced motion")
        }
    }
}
#endif

File: androidApp/src/debug/kotlin/it/danielebufarini/trenify/design/TrenifyComponentPreviews.kt
package it.danielebufarini.trenify.design

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import it.danielebufarini.trenify.design.component.*
import it.danielebufarini.trenify.design.theme.TrenifySpacing
import it.danielebufarini.trenify.design.theme.TrenifyTheme

/** Debug source set only; never registered in production navigation. */
@Preview(name = "Light / component family", showBackground = true, widthDp = 390, heightDp = 1600)
@Preview(name = "Large text / narrow", fontScale = 2f, showBackground = true, widthDp = 320, heightDp = 1800)
@Composable
private fun LightComponents() = Components(dark = false)

@Preview(name = "Dark / component family", showBackground = true, widthDp = 390, heightDp = 1600)
@Composable
private fun DarkComponents() = Components(dark = true)

@Composable
private fun Components(dark: Boolean) {
    TrenifyTheme(darkTheme = dark, reduceMotion = true) {
        Surface(color = TrenifyTheme.colors.background) {
            Column(Modifier.verticalScroll(rememberScrollState()).padding(TrenifySpacing.screenHorizontal),
                verticalArrangement = Arrangement.spacedBy(TrenifySpacing.xl)) {
                TrenifySectionHeader("Component family", subtitle = "Deterministic railway display fixtures")
                TrenifySegmentedControl(listOf(TrenifySegment("journey", "Journey"), TrenifySegment("train", "Train")), "journey", {})
                TrenifyStationField("From", "San Benedetto del Tronto Porto d’Ascoli", "Choose station", {}, selected = true, onClear = {})
                TrenifyStationField("To", null, "Choose station", {}, focused = true)
                TrenifyStationField("Station", "Roma Termini", "Choose station", {}, enabled = false)
                TrenifyPrimaryAction("Search trains", {})
                TrenifyPrimaryAction("Search trains", {}, loading = true)
                TrenifySecondaryAction("Change selection", {}, enabled = false)
                TrenifyJourneyCard("Trenitalia · Frecciarossa 9516", "14:10", "17:19",
                    "Milano Centrale", "Roma Termini", "3 h 09 min", "Direct", {},
                    statusTone = TrenifyStatusTone.OnTime, statusLabel = "On time", platformLabel = "Scheduled platform 8")
                TrenifyStatusPill(TrenifyStatusTone.Delayed, "Delayed +12 min")
                TrenifyStatusPill(TrenifyStatusTone.Cancelled, "Cancelled")
                TrenifyStatusPill(TrenifyStatusTone.Arrived, "Arrived")
                TrenifyStatusPill(TrenifyStatusTone.Warning, "Service warning")
                TrenifyStatusPill(TrenifyStatusTone.Unknown, "Status unavailable")
            }
        }
    }
}
